### PR TITLE
Remove yet more of the specific methods for reading and writing NamedWriteables

### PR DESCRIPTION
**Pull request by nik9000**
This is just more cleanup related to #17682.

### CODE REVIEW

**Comment by nik9000** [comment]
@ywelsch this is the second half of #17838, it removes the last of the specialized readers.

File: core/src/main/java/org/elasticsearch/index/query/SpanFirstQueryBuilder.java [hunk 1]
```diff
@@ -67,13 +67,13 @@ public SpanFirstQueryBuilder(SpanQueryBuilder<?> matchBuilder, int end) {
      */
     public SpanFirstQueryBuilder(StreamInput in) throws IOException {
         super(in);
-        matchBuilder = (SpanQueryBuilder<?>)in.readQuery();
+        matchBuilder = (SpanQueryBuilder<?>)in.readNamedWriteable(QueryBuilder.class);
```
Comment (ywelsch) [inline core/src/main/java/org/elasticsearch/index/query/SpanFirstQueryBuilder.java]: nit: space missing between cast and value

**Comment by ywelsch** [comment]
Thanks for the cleanup @nik9000. LGTM.

**Comment by nik9000** [comment]
Thanks for reviewing @ywelsch !

**Pull request merged**